openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go

```diff
@@ -152,26 +174,38 @@ func (a *managementCPUsOverride) Admit(ctx context.Context, attr admission.Attri
 	}
 
 	if !a.waitForSyncedStore(time.After(timeToWaitForCacheSync)) {
-		return admission.NewForbidden(attr, fmt.Errorf("%s node and namespace caches not synchronized", PluginName))
+		return admission.NewForbidden(attr, fmt.Errorf("%s node or namespace or infra config cache not synchronized", PluginName))
+	}
+
+	if a.isSNOCluster == nil {
+		clusterInfra, err := a.infraConfigLister.Get(infraClusterName)
+		if err != nil {
+			return admission.NewForbidden(attr, err) // can happen due to informer latency
+		}
+
+		a.isSNOCluster = pointer.BoolPtr(
```

Comment:
this races. You can't write unprotected state.

Reply:
Why do you store this value at all? It is from a lister, i.e. has `O(1)` lookup as there is only one object. So no need to cache the value. Get rid of it inicluding the race.

Reply:
Good point! Thanks!